fakestorage/json_response.go

```diff
@@ -63,5 +65,9 @@ func errToJsonResponse(err error) jsonResponse {
 	if errors.As(err, &pathError) && pathError.Err == syscall.ENAMETOOLONG {
 		status = http.StatusBadRequest
 	}
+	var backendError backend.Error
+	if errors.As(err, &backendError) && backendError == backend.PreConditionFailed {
```

Comment:
we can simplify this check:
```suggestion
	if err == backend.PreConditionFailed {
```